Slider/Assets/Scripts/UI/UISpriteSwapper.cs

```diff
@@ -0,0 +1,22 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+public class UISpriteSwapper : FlashWhiteUI
+{
+    [SerializeField] private Sprite on;
+    [SerializeField] private Sprite off;
+    [SerializeField] private Image image;
+    private ArtifactTileButton button;
+
+    protected override void Awake()
+    {
+        base.Awake();
+        button = GetComponentInParent<ArtifactTileButton>();
+    }
+    
+    private void Update() => image.enabled = button.TileIsActive;
```

Comment:
these being arrow functions is so goofy lmao also how is this different from `SpriteSwapper.cs`/what use case

Reply:
need the `button.tileIsActive` check similar to power line ui or else it will always show on UI even when u dont have tile